src/streamlink/plugins/niconicochannelplus.py

```diff
@@ -0,0 +1,67 @@
+"""
+$description NicoNico Channel Plus (nicochannel+) is a new feature of Niconico Channel.
+$url nicochannel.jp
+$type vod
+$account Not required. Full-length videos are accessible.
+$notes Downloading non-free content is not recommended.
+"""
+
+import json
+import logging
+import re
+
+from streamlink.plugin import Plugin, pluginmatcher
+from streamlink.plugin.api import HTTPSession
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(
+    re.compile(
+        r'^https?://nicochannel\.jp/(?P<channel>[a-z0-9_-]+)/video/(?P<id>sm[a-zA-Z0-9]+)$'
+    )
+)
+class NicoNicoChannelPlus(Plugin):
+    def get_video_page_info(self, http: HTTPSession, video_id: str) -> dict:
+        video_page_json = json.loads(
+            http.get(
+                url=f'https://nfc-api.nicochannel.jp/fc/video_pages/{video_id}',
+            ).content
+        )['data']['video_page']
+
+        return video_page_json
+
+    def get_master_playlist_url(self, http: HTTPSession, video_id: str) -> str:
+        session_id = json.loads(
+            http.post(
+                url=f'https://nfc-api.nicochannel.jp/fc/video_pages/{video_id}/session_ids',
+                data=str({}).encode('ascii'),
+                headers={
+                    'content-type': 'application/json',
+                }
+            ).content
+        )['data']['session_id']
+
+        return f'https://hls-auth.cloud.stream.co.jp/auth/index.m3u8?session_id={session_id}'
+
+    def _get_streams(self):
+        self.id = self.match.group('id')
+        self.author = self.match.group('channel')
+
+        video_info = self.get_video_page_info(self.session.http, self.id)
+        playlist_url = self.get_master_playlist_url(self.session.http, self.id)
+
+        self.title = video_info['title']
+
+        log.info(f'ID:      {self.id}')
+        log.info(f'Channel: {self.author}')
+        log.info(f'Title:   {self.title}')
```

Comment:
That would be completely unnecessary

Reply:
Hmm, "id" and "author" are indeed unnecessary since they are in the URL without any encoding; what about the title? maybe I can put it into `log.debug` ?

Reply:
> ... what about the title? maybe I can put it into `log.debug` ?

3d415c2024d5c6370bb219fe7baabbab70a66f6a

Reply:
To be honest, even having the title is (I would have thought) not that useful for debugging purposes...

Reply:
@mkbloke, thanks for your review.

Well, the title is indeed useless for debugging, but a string extracted from JSON and does provide something meaningful. Since the URL is not fully semantic, users are able to check if they are streaming/downloading the correct video by identifying the title.

I would even like to change it to `log.info` ; it seems OK to output this type of information in Streamlink:

https://github.com/streamlink/streamlink/blob/ca4fff1453fbc10dcd583c8a88c922fb7ad16fd1/src/streamlink/plugins/crunchyroll.py#L390

https://github.com/streamlink/streamlink/blob/ca4fff1453fbc10dcd583c8a88c922fb7ad16fd1/src/streamlink/plugins/htv.py#L52

<br>

Your opinion?

Reply:
I think given that URLs are usually either copied and pasted or occasionally sent via something like the "Open With" browser extension, there is probably not much value in outputting the title.

I have found since joining Streamlink that the other devs generally lean toward keeping the text output (info level) to a minimum and only using it if it's really necessary.